### PR TITLE
[Hotfix] - Add TLS settings for redis plugin

### PR DESCRIPTION
**Pull request by xwellingtonx**
The Redis plugin was not exposing the TLS settings on the UI, which makes the usage of Azure Redis Cache impossible. This PR intention is to fix that.

Obs: This Redis plugin might have to be reviewed again, since at the current version it won't work with Redis clusters.

Video: 
[20240807-0928-21.6391904.zip](https://github.com/user-attachments/files/16522621/20240807-0928-21.6391904.zip)

### CODE REVIEW

File: plugins/packages/redis/lib/manifest.json
Comment (akshaysasidrn) [inline plugins/packages/redis/lib/manifest.json:81]: Since it is only added for tls option, can we rethink this part? Connection options are generally spread at the root level. One approach is just prefix it with `tls_*`. 

But instead of that, I think we can do it like how we've done it for other datasources. Please have a look at the manifest file for Postgres datasource for reference.
We can have dropdown for the type of tls selection with `None` and `CA certificate` as an option. That will solve it, isn't it?

**Review by akshaysasidrn — CHANGES_REQUESTED** [review]

File: plugins/packages/redis/lib/manifest.json
Comment (xwellingtonx) [reply plugins/packages/redis/lib/manifest.json:81]: @akshaysasidrn I have followed the postgress plugin, the TLS options for ioredis do not support certificates, at least if you are not using the Redis.Cluster, which it's the case here

File: plugins/packages/redis/lib/index.ts
Comment (xwellingtonx) [reply plugins/packages/redis/lib/index.ts]: @akshaysasidrn fixed

File: plugins/packages/redis/lib/manifest.json
Comment (xwellingtonx) [reply plugins/packages/redis/lib/manifest.json:81]: Could you give me some example of what you mean by prefixing?

Comment (xwellingtonx) [reply plugins/packages/redis/lib/manifest.json:81]: I'm not quite sure what you mean here, the Connection Options on postgress plugin is shown all the time, it does not control if it will show or not the Connection options.

Comment (xwellingtonx) [reply plugins/packages/redis/lib/manifest.json]: I have changed it, but I don't understand why I need to add the common fields under the tls_certificate section, it should be outside of this section, but if I let outside it doesn't render the UI.


https://github.com/user-attachments/assets/402005c6-a484-49f3-ba84-63188fbf90bd

Comment (xwellingtonx) [reply plugins/packages/redis/lib/manifest.json]: @akshaysasidrn I've understood now. But we should use the connection options inside the TLS section, because we can or not inform HOST/PORT and Family, those parameters are not mandatory but can me informed if needed

Comment (akshaysasidrn) [reply plugins/packages/redis/lib/manifest.json]: @xwellingtonx Is that something that blocks you at the moment? If not, we can add it later iteratively.

Comment (xwellingtonx) [reply plugins/packages/redis/lib/manifest.json]: @akshaysasidrn No, it's not! I can remove it so, I just added in case someone else would need after the deployment. I'm going to remove it since I need this be deployed so it unblock my APP

Comment (akshaysasidrn) [reply plugins/packages/redis/lib/manifest.json]: Cool lets go ahead then 🙌🏾

Comment (xwellingtonx) [reply plugins/packages/redis/lib/manifest.json]: @akshaysasidrn  all done.

Comment (akshaysasidrn) [reply plugins/packages/redis/lib/manifest.json]: @xwellingtonx thanks, will have this tested and then take it forward for a release.

**Comment by akshaysasidrn** [comment]
@xwellingtonx We are testing it out and found an issue with ioredis package. When the TLS connection fails, it keeps on retrying to connect to redis infinitely.
I tried it against a [mock redis setup](https://github.com/akshaysasidrn/TjMocks/tree/main/redis) for these config to verify. We'd need to fix this before merging. Let me look into this.


redis

**Comment by xwellingtonx** [comment]
> @xwellingtonx We are testing it out and found an issue with ioredis package. When the TLS connection fails, it keeps on retrying to connect to redis infinitely. I tried it against a [mock redis setup](https://github.com/akshaysasidrn/TjMocks/tree/main/redis) for these config to verify. We'd need to fix this before merging. Let me look into this.
> 
> redis [ redis ](https://private-user-images.githubusercontent.com/11629675/357418278-0f4f172e-fada-4bf1-9261-234c7d37b2a5.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM1NTU2MzYsIm5iZiI6MTcyMzU1NTMzNiwicGF0aCI6Ii8xMTYyOTY3NS8zNTc0MTgyNzgtMGY0ZjE3MmUtZmFkYS00YmYxLTkyNjEtMjM0YzdkMzdiMmE1LmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODEzVDEzMjIxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTEwNWVjMzRiYTRmMmI3YmVlNTVhMzljOWMzYWUxNzEwNTc2NGQxNDljZDcyZjliOGVlNGFkNWNmYTkzYjIyNDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Yll7BGsLUEo1DShjgD8k5sVIF4kAu-suVoG_UQ0ydis) [ ](https://private-user-images.githubusercontent.com/11629675/357418278-0f4f172e-fada-4bf1-9261-234c7d37b2a5.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM1NTU2MzYsIm5iZiI6MTcyMzU1NTMzNiwicGF0aCI6Ii8xMTYyOTY3NS8zNTc0MTgyNzgtMGY0ZjE3MmUtZmFkYS00YmYxLTkyNjEtMjM0YzdkMzdiMmE1LmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODEzVDEzMjIxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTEwNWVjMzRiYTRmMmI3YmVlNTVhMzljOWMzYWUxNzEwNTc2NGQxNDljZDcyZjliOGVlNGFkNWNmYTkzYjIyNDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Yll7BGsLUEo1DShjgD8k5sVIF4kAu-suVoG_UQ0ydis)

Hi @akshaysasidrn I noted this even when not using SSL, so I don't think is related to my changes, most likely it's due to how TestConnection function is called by Tooljet.

File: plugins/packages/redis/lib/manifest.json
Comment (xwellingtonx) [reply plugins/packages/redis/lib/manifest.json]: @akshaysasidrn this is done now

**Comment by akshaysasidrn** [comment]
> Hi @akshaysasidrn I noted this even when not using SSL, so I don't think is related to my changes, most likely it's due to how TestConnection function is called by Tooljet.

@xwellingtonx Thanks for pointing it out to me. So we need to terminate the client if it fails to establish the connection. That is missed currently. Can you also add this change in? Something like:

```js
 async testConnection(sourceOptions: SourceOptions): Promise<ConnectionTestResult> {
    const client = await this.getConnection(sourceOptions);

    try {
      await client.ping();
    } catch (err) {
      client.disconnect();
      throw new QueryError('Connection could not be established', err.message, {});
    }

    return {
      status: 'ok',
    };
  }
  ```

**Comment by xwellingtonx** [comment]
> ```js
> try {
>       await client.ping();
>     } catch (err) {
>       client.disconnect();
>       throw new QueryError('Connection could not be established', err.message, {});
>     }
> ```

@akshaysasidrn this is now done.

**Review by akshaysasidrn — APPROVED** [review]

**Comment by xwellingtonx** [comment]
Hi @akshaysasidrn , any update of the testing outcome on this?

**Comment by akshaysasidrn** [comment]
Hey @xwellingtonx , it looks good. We are planning to release this at the earliest for LTS.

**Pull request merged**
